vim-patch:8.1.{837,1180,1194,1203,1207,1209}

janlazo opened the pull request:
**vim-patch:8.1.0837: timer interrupting cursorhold and mapping not tested**

Problem:    Timer interrupting cursorhold and mapping not tested.
Solution:   Add tests with timers. (Ozaki Kiichi, closes vim/vim#3871)
https://github.com/vim/vim/commit/26d982185e21398738a9c688429c0a1840d7c9c3

**vim-patch:8.1.1180: Vim script debugger tests are old style**
Problem:    Vim script debugger tests are old style.
Solution:   Turn into new style tests. (Yegappan Lakshmanan, closes vim/vim#4259)
vim/vim@113bf06

**vim-patch:8.1.1194: typos and small problems in source files**
Problem:    Typos and small problems in source files.
Solution:   Small fixes.
vim/vim@ad3ec76

**vim-patch:8.1.1203: some autocmd tests are old style**
Problem:    Some autocmd tests are old style.
Solution:   Turn the tests into new style. (Yegappan Lakshmanan, closes vim/vim#4295)
vim/vim@69ea587
18fddad

**vim-patch:8.1.1207: some compilers give warning messages**
Problem:    Some compilers give warning messages.
Solution:   Initialize variables, change printf() argument. (Christian Brabandt, closes vim/vim#4305)
vim/vim@1f3601e

**vim-patch:8.1.1209: clever compiler warns for buffer being too small**
Problem:    Clever compiler warns for buffer being too small.
Solution:   Make the buffer bigger (even though it's not really needed).
vim/vim@5431589

janlazo committed:
vim-patch:8.1.0837: timer interrupting cursorhold and mapping not tested

Problem:    Timer interrupting cursorhold and mapping not tested.
Solution:   Add tests with timers. (Ozaki Kiichi, closes vim/vim#3871)
https://github.com/vim/vim/commit/26d982185e21398738a9c688429c0a1840d7c9c3

janlazo committed:
vim-patch:8.1.1194: typos and small problems in source files

Problem:    Typos and small problems in source files.
Solution:   Small fixes.
https://github.com/vim/vim/commit/ad3ec76bb8030b9a1d3f0a49c374d0de2383b977

janlazo committed:
vim-patch:8.1.1203: some autocmd tests are old style

Problem:    Some autocmd tests are old style.
Solution:   Turn the tests into new style. (Yegappan Lakshmanan, closes vim/vim#4295)
https://github.com/vim/vim/commit/69ea587289b03e23a9fb96adffd6e8173cbc5896

janlazo commented on src/nvim/testdir/test_autocmd.vim:
```diff
@@ -1401,3 +1401,205 @@ func Test_autocmd_once()
+    au!
+    au BufNewFile      *.c    read Xtest.c
+    au FileAppendPre   *.out  '[,']s/new/NEW/
+    au FileAppendPost  *.out  !cat Xtest.c >> test.out
```
Test always passes if `test.out` is deleted. `test_autocmd.vim` passes if it is the only test that runs even if `test.out` wasn't deleted. Why?

justinmk replied:
I don't follow your question. Do mean, you delete `test.out` manually before running the test?

justinmk replied:
Hmm, I see the same behavior on my local env.

justinmk replied:
Vim deletes `test.out` files before running each old-old-test:

https://github.com/vim/vim/blob/8dc4c7292364d31cfb8afa8a67984b78b657b192/src/testdir/Makefile#L97

I think we are missing that for the old-old-tests.

justinmk replied:
We delete `test.out` files before each run, as Vim does:

https://github.com/neovim/neovim/blob/9d77a0768697aced470c72ba71d0e04cf45f0fd0/src/nvim/testdir/Makefile#L139

So this may just be an accident of ordering. Our test runner is automated, so the order may be different than in Vim.

janlazo committed:
vim-patch:8.1.1180: Vim script debugger tests are old style

Problem:    Vim script debugger tests are old style.
Solution:   Turn into new style tests. (Yegappan Lakshmanan, closes vim/vim#4259)
https://github.com/vim/vim/commit/113bf0672b114af5800fa642bcd09f674758e5d5

janlazo committed:
oldtests: pass Test_ReadWrite_Autocmds()

janlazo committed:
vim-patch:8.1.1207: some compilers give warning messages

Problem:    Some compilers give warning messages.
Solution:   Initialize variables, change printf() argument. (Christian
            Brabandt, closes vim/vim#4305)
https://github.com/vim/vim/commit/1f3601e92e7fd2813b9541580d6d9649c802eb58

janlazo committed:
vim-patch:8.1.1209: clever compiler warns for buffer being too small

Problem:    Clever compiler warns for buffer being too small.
Solution:   Make the buffer bigger (even though it's not really needed).
https://github.com/vim/vim/commit/5431589d25e73892fcf7ad1eaca53f742c1c9303

justinmk commented:
This should fix it:

```diff
diff --git a/src/nvim/testdir/Makefile b/src/nvim/testdir/Makefile
index fac27aa34620..6ce49e7596e5 100644
--- a/src/nvim/testdir/Makefile
+++ b/src/nvim/testdir/Makefile                                                                                                                              
@@ -164,5 +164,6 @@ newtestssilent: $(NEW_TESTS)                                                                                                             
-                                                                                                                                                            
 %.res: %.vim .gdbinit                                                                                                                                       
 >      @echo "[OLDTEST] Running" $*                                                                                                                         
+>      @rm -rf $*.failed test.ok $(RM_ON_RUN)                                                                                                               
 >      @mkdir -p $(TMPDIR)                                                                                                                                  
 >      @/bin/sh runnvim.sh $(ROOT) $(NVIM_PRG) $* $(RUN_VIMTEST) -u NONE -S runtest.vim $*.vim
```

justinmk committed:
test/old: remove test.out files

Do this for new-style tests too, because they run after the old-style
tests which don't clean up their temp files.

Reverts ebd251c
